cognite/neat/core/rules/importer/owl2excel.py

```diff
@@ -0,0 +1,358 @@
+
+from cognite.neat.core.rules import _exceptions, load_rules_from_excel_file
+from cognite.neat.core.utils.utils import generate_exception_report, get_namespace, remove_namespace
+
+
+def owl2excel(owl_filepath: Path, excel_filepath: Path = None, validate_results: bool = True):
+    """Convert owl ontology to transformation rules serialized as Excel file.
+
+    Parameters
+    ----------
+    owl_filepath : Path
+        Path to OWL ontology
+    filepath : Path
+        Path to save transformation rules, defaults to None
+    validate_results : bool, optional
+        Whether to validate generated Excel file and create validation report, by default True
+    """
+
+    owl_filepath = Path(owl_filepath)
+    if excel_filepath:
+        excel_filepath = Path(excel_filepath)
+    else:
+        excel_filepath = owl_filepath.parent / "transformation_rules.xlsx"
+
+    graph = Graph()
+    try:
+        graph.parse(owl_filepath)
+    except Exception as e:
+        raise Exception(f"Could not parse owl file: {e}") from e
+
+    # bind key namespaces
+    graph.bind("owl", OWL)
+    graph.bind("rdf", RDF)
+    graph.bind("rdfs", RDFS)
+    graph.bind("dcterms", DCTERMS)
+    graph.bind("dc", DC)
+
+    writer = pd.ExcelWriter(excel_filepath)
+
+    _parse_owl_metadata_df(graph).to_excel(writer, sheet_name="Metadata", header=False)
+    _parse_owl_classes_df(graph).to_excel(writer, sheet_name="Classes", index=False, header=False)
+    _parse_owl_properties_df(graph).to_excel(writer, sheet_name="Properties", index=False, header=False)
+
+    writer.close()
+
+    if validate_results:
+        _validate_excel_file(excel_filepath)
+
+
+def _validate_excel_file(excel_filepath: Path):
+    _, validation_errors, validation_warnings = load_rules_from_excel_file(excel_filepath, return_report=True)
+
+    report = ""
+    if validation_errors:
+        warnings.warn(
+            _exceptions.Warning1().message,
+            category=_exceptions.Warning1,
+            stacklevel=2,
+        )
+        report = generate_exception_report(validation_errors, "Errors")
+
+    if validation_warnings:
+        warnings.warn(
+            _exceptions.Warning2().message,
+            category=_exceptions.Warning2,
+            stacklevel=2,
+        )
+        report += generate_exception_report(validation_warnings, "Warnings")
+
+    if report:
+        with open(excel_filepath.parent / "report.txt", "w") as f:
+            f.write(report)
+
+
+# TODO: these are to be read from Metadata pydantic model
+PARSING_METADATA = {
+    "header": (
+        "namespace",
+        "prefix",
+        "dataModelName",
+        "cdfSpaceName",
+        "version",
+        "isCurrentVersion",
+        "created",
+        "updated",
+        "title",
+        "description",
+        "creator",
+        "contributor",
+        "rights",
+        "license",
+    )
+}
+
+
+def _parse_owl_metadata_df(graph: Graph, parsing_config: dict = PARSING_METADATA) -> pd.DataFrame:
```

Comment:
The PARSING_METADATA is mutable. Suggest a solution such as:
```suggestion
def _parse_owl_metadata_df(graph: Graph, parsing_config: dict = None) -> pd.DataFrame:
    parsing_config = parsing_config or PARSING_METADATA
```

Reply:
Note the above is also risky, as if the function does any changes to `pasing_config` those will be present the next time the function is called. So the safest version would be something like this
```suggestion
def _parse_owl_metadata_df(graph: Graph, parsing_config: dict = None) -> pd.DataFrame:
    if parsing_config is None:
        parsing_config = _create_default_parsing_config()
```